openedx/core/djangoapps/content/search/management/commands/reindex_studio.py

```diff
@@ -0,0 +1,140 @@
+"""
+Command to build or re-build the search index for courses (in Studio, i.e. Draft
+mode), in Meilisearch.
+
+See also cms/djangoapps/contentstore/management/commands/reindex_course.py which
+indexes LMS (published) courses in ElasticSearch.
+"""
```

Comment:
I'm not sure what standard to hold this module to. As a command intended for developers to be able to bootstrap and test local dev envs with search functionality, this is fine. I might ask for small features like being able to specify a specific library or course. As a migration, I'd have more requests w.r.t. scaling, logging, debug output, etc.

Could you maybe indicate that it's experimental and not safe to run in production envs here?

Reply:
This command creates a brand new index, configures it, populates it, then swaps it to become the active index. It wouldn't make sense to run it for a single course, because it would erase all other courses from the resulting index. But it could make sense to have an option to skip populating it with content (for initial setup on huge instances), and another command that would reindex a single course/library (into the active index, without creating a new index), to give more control.

I have indicated it's experimental and it now requires the `--experimental` flag.

https://github.com/openedx/edx-platform/blob/ec0c4a875d7bc4766ff02c50a5e36d8ec6cd3237/openedx/core/djangoapps/content/search/management/commands/reindex_studio.py#L33-L48